tests/xen/test_misc.py

```diff
@@ -0,0 +1,22 @@
+import logging
+import pytest
+import re
+
+# Misc tests for Xen host
+#
+# Requirements:
+# - XCP-ng host
+
+# Check for a valid "Latest ChangeSet" trace in 'xl dmesg'
+def test_xen_changeset(host):
+    changeset = host.ssh(['xl', 'dmesg', '|', 'grep', '"Latest ChangeSet"'])
+    regexp = r'.*Latest ChangeSet:\s*(([^,]+),.*)'
+
+    m = re.match(regexp, changeset)
+    assert m is not None, "'Latest ChangeSet' should be found in 'xl dmesg'"
+
+    full_changeset, git_sha = m.groups()
+    logging.info(f"Xen Latest ChangeSet: {full_changeset}")
+
+    m = re.match('[0-9a-fA-F]+', git_sha)
```

Comment:
Add `r` as in the first pattern?
```suggestion
    m = re.match(r'[0-9a-fA-F]+', git_sha)
```

Reply:
`r''` is really only useful when there is a `\` in the regexp

Reply:
Right but still, it's a pattern so...

Reply:
A pattern does not *have to* be a *raw string* - `r` does not stand for *regexp* ;)